R/OMLData.R

```diff
@@ -201,9 +200,25 @@ OMLData = R6Class("OMLData",
       }
       if (self$parquet) {
         path = self$parquet_path
-        private$.backend = mlr3db::as_duckdb_backend(path)
+        backend = mlr3db::as_duckdb_backend(path)
+        if (!test_names(backend$colnames, type = "strict")) {
+          new = make.names(backend$colnames)
+          if (length(unique(new)) != length(new)) {
+            stopf("No unique names after conversion.")
+          }
+          # This code is a little hacky. The reason is that DataBackendRename is not exported
+          # in mlr3 and only accessible via DataBackendRename (mlr3 version 0.14).
+          # This can be changed when it is exported in mlr3.
+          private$.backend = withr::with_options(list(mlr3.allow_utf8_names = TRUE),
```

Comment:
agreed, this is hacky. i can export the backend.

also: would it be possible to just create a view with modified column names in duckdb? maybe this could be included in mlr3db.

Reply:
either way, this step should only be executed iff `make.names()` performed any renames.

Reply:
> agreed, this is hacky. i can export the backend.

I leave it until it is exported. 


> also: would it be possible to just create a view with modified column names in duckdb? maybe this could be included in mlr3db.

Ok, created a issue here: https://github.com/mlr-org/mlr3db/issues/24

Reply:
> either way, this step should only be executed iff `make.names()` performed any renames.

But I check this with `test_names(..., type = "strict")` (?)